src/fomo.jl

```diff
@@ -346,10 +346,7 @@ function SolverCore.solve!(
     step_underflow = x == c # step addition underfow on every dimensions, should happen before solver.α == 0
     ΔTk = ((oneT - βmax) * norm_∇fk^2 + βmax * mdot∇f) * λk # = dot(d,∇fk) * λk with momentum, ‖∇fk‖²λk without momentum
     fck = obj(nlp, c)
-    if fck == -Inf
-      set_status!(stats, :unbounded)
-      break
-    end
+    unbounded = isinf(fck)
```

Comment:
I would go for a test like `fx < min(-one(T), f0) / eps(T)` instead of testing Inf.